### PR TITLE
Converting all the Checkbox in Preference section into Toggle Switches

### LINKED ISSUE #10776: [Suggestion] Replace all the checkboxes with the toggle switches in the Preferences section
**Is your feature request related to a problem? Please describe.**
Material uses toggles instead of checkboxes for Preferences/Settings screens
https://material.io/archive/guidelines/patterns/settings.html#settings-labels-secondary-text

**Describe the solution you'd like**
I would like to replace all checkboxes with toggle switches

Toggle Switch

image

Checkboxes

image

### PR DESCRIPTION
**Pull request by UvrajSB**
## Purpose / Description
Converting all the Checkbox in the Preference section into Toggle Switches

## Fixes
Fixes #10776 

## Approach
CheckBox and Switch are two custom selection elements provided by the preference library and they have an identical implementation, its just CheckBox has CheckBoxPreferences and Switch has SwitchPreferences, and the two elements being two-option having elements have identical possible states as well.

## How Has This Been Tested?

Tested by my physical device: Realme 8i

https://user-images.githubusercontent.com/73571511/163495066-f607e85e-0278-4d1f-bd82-3ce3b31104df.mp4

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
if you type `Fixes #issue` then it auto-links the PR and the issue and closes one when the other is closed. I've edited the main post to do this:

```
## Fixes
Fixes #10776 
```

**Comment by UvrajSB** [comment]
> if you type `Fixes #issue` then it auto-links the PR and the issue and closes one when the other is closed. I've edited the main post to do this:
> 
> ```
> ## Fixes
> Fixes #10776 
> ```

I will keep this in mind next time, thank you

**Review by david-allison — DISMISSED** [review]
* Custom Sync Server & Advanced statistics aren't changed

Looks good otherwise

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt [hunk 1]
```diff
@@ -259,21 +259,21 @@ class Preferences : AnkiActivity() {
...
                         USE_CURRENT -> (pref as ListPreference).setValueIndex(if (col.get_config("addToCur", true)!!) 0 else 1)
                         AUTOMATIC_ANSWER_ACTION -> (pref as ListPreference).setValueIndex(col.get_config(AutomaticAnswerAction.CONFIG_KEY, 0.toInt())!!)
                         NEW_SPREAD -> (pref as ListPreference).setValueIndex(col.get_config_int("newSpread"))
                         DAY_OFFSET -> (pref as SeekBarPreferenceCompat).value = getDayOffset(col)
-                        PASTE_PNG -> (pref as CheckBoxPreference).isChecked = col.get_config("pastePNG", false)!!
+                        PASTE_PNG -> (pref as SwitchPreference).isChecked = col.get_config("pastePNG", false)!!
                         NEW_TIMEZONE_HANDLING -> {
-                            val checkBox = pref as CheckBoxPreference
-                            checkBox.isChecked = col.sched._new_timezone_enabled()
+                            val Switch = pref as SwitchPreference
```
Comment (TacoTheDank) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:272]: ``Switch`` should not be uppercase, as this is not Kotlin convention.

File: AnkiDroid/src/main/res/xml/preferences_advanced.xml [hunk 2]
```diff
@@ -138,19 +138,19 @@
...
-            <CheckBoxPreference
+            <SwitchPreference
                 android:defaultValue="false"
                 android:key="convertFenText"
                 android:summary="@string/convert_fen_text_summ"
                 android:title="@string/convert_fen_text" />
-            <CheckBoxPreference
-                android:id="@+id/scrolling_buttons_checkbox"
+            <SwitchPreference
+                android:id="@+id/scrolling_buttons_Switch"
```
Comment (TacoTheDank) [inline AnkiDroid/src/main/res/xml/preferences_advanced.xml:147]: This id isn't used anywhere in the project from what I can see. It can probably just be removed. Same with ``double_scrolling_Switch``.

**Review by TacoTheDank — COMMENTED** [review]
Looks good overall. Just some comments...

File: AnkiDroid/src/main/res/xml/preferences_general.xml [hunk 3]
```diff
@@ -22,8 +22,9 @@
 
 <!--  General Preferences -->
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (Akshay0701) [inline AnkiDroid/src/main/res/xml/preferences_general.xml:25]: you added the app namespaces, any reason? 
do we use the `app: ___` in this file?

**Review by Akshay0701 — COMMENTED** [review]
Thanks for the contribution @UvrajSB 
just one doubt regrading xmlns: app, otherwise LGTM

File: AnkiDroid/src/main/res/xml/preferences_advanced.xml (hunk 2 above)
Comment (UvrajSB) [reply AnkiDroid/src/main/res/xml/preferences_advanced.xml:147]: image
They are used for these, perhaps you didn't scroll.

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt (hunk 1 above)
Comment (UvrajSB) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:272]: Yeah, thanks changed

File: AnkiDroid/src/main/res/xml/preferences_general.xml (hunk 3 above)
Comment (UvrajSB) [reply AnkiDroid/src/main/res/xml/preferences_general.xml]: I was trying to change the color of the toggle switches but wasn't able to find a way through the preferences library yet and then I observed that even the checkboxes we were using didn't have the Ankidroid blue color

File: AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt [hunk 4]
```diff
@@ -57,11 +56,9 @@ class PreferencesTest : RobolectricTest() {
...
-    val instance: Preferences
-        get() {
-            val preferences = Preferences()
-            preferences.attachBaseContext(targetContext)
-            return preferences
+    val instance by lazy {
+        Preferences().apply {
+            attachBaseContext(targetContext)
         }
+    }
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt:63]: This cleanup is unrelated to your main PR - it's trivial enough that it is okay to include it, so no action is required
However! In general only do in a PR what is related to the actual issue being closed for the PR. There is no problem creating multiple PRs

Comment (UvrajSB) [reply AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt:63]: Yeah, I understand. That cleanup's commit was unintentionally added as the first time I made the PR, I forgot to drop the commit while rebasing.
Apologies for that, I will be more careful next time

**Review by mikehardy — APPROVED** [review]
LGTM - I'm happy to see this merge as soon as all other comments are addressed and commenters sign off
(incidentally, :wave: @TacoTheDank ! good to see you)

**Review by Akshay0701 — APPROVED** [review]
LGTM, Thanks!

File: AnkiDroid/src/main/res/xml/preferences_general.xml (hunk 3 above)
Comment (mikehardy) [reply AnkiDroid/src/main/res/xml/preferences_general.xml]: (or the styles method perhaps)

File: AnkiDroid/src/main/res/xml/preferences_advanced.xml (hunk 2 above)
Comment (TacoTheDank) [reply AnkiDroid/src/main/res/xml/preferences_advanced.xml:147]: Yes, I'm aware of that. What I mean is the ids themselves (not the switches) aren't used anywhere in the project. They are not referenced anywhere. Therefore, these ids serve no purpose and can just be removed.

File: AnkiDroid/src/main/res/xml/preferences_general.xml (hunk 3 above)
Comment (UvrajSB) [reply AnkiDroid/src/main/res/xml/preferences_general.xml]: Hey Mike!
Thanks for it but I tried both of the methods the tint one as well as the styles method but they are not really working for me.
As you can see and from what I have observed that the toggle switches we are using are not the ones they are discussing about (standard switch widgets).
We are using the preferences library that uses SwitchPreference for creating the widgets and the attributes for the std switches are not working in it.

Comment (UvrajSB) [reply AnkiDroid/src/main/res/xml/preferences_general.xml]: Otherwise, I don't see a reason why the person who actually implemented it first would have got satisfied with the dark green color of the checkboxes if there were no issues.
Anyways I would insist that we should merge these many changes and make an issue to change the color of headings and switches of the preferences section.
Even if this wouldn't has been here, we still required that issue as in past implementation the colors were mismatching only.

File: AnkiDroid/src/main/res/xml/preferences_advanced.xml (hunk 2 above)
Comment (UvrajSB) [reply AnkiDroid/src/main/res/xml/preferences_advanced.xml]: Oh my bad, I will remove that.
Thanks for pointing out

**Review by david-allison — APPROVED** [review]

**Comment by david-allison** [comment]
I think I was holding this up, I'm more than happy, looks great when I tested on my phone, thanks!!!

File: AnkiDroid/src/main/res/xml/preferences_general.xml (hunk 3 above)
Comment (mikehardy) [reply AnkiDroid/src/main/res/xml/preferences_general.xml]: Agreed, not a regression - neat github trick, if you look at the three dots menu on the top right of every comment, you can make a new issue from a comment and it gets some auto-created connections to this as background context. If you could do that about the lack of theme colors (just change the title of the issue to be reasonable) that would help track it. It's a good issue by itself.
